planner/core/logical_plan_builder.go

```diff
@@ -3152,6 +3311,127 @@ func (b *PlanBuilder) checkOnlyFullGroupByWithGroupClause(p LogicalPlan, sel *as
 	return nil
 }
 
+func (b *PlanBuilder) checkOrderByPart1(ctx context.Context, sel *ast.SelectStmt, p LogicalPlan) error {
+	if sel.GroupBy != nil {
+		// check only-full-group-by 4 order-by clause with group-by clause
+		// Step#1: whether order by item is in the select field list
+		// Step#2: whether order by item is in the group by list
+		// Step#3: whether order by item is in the FD closure of group-by & select list items.
+		// Since FD hasn't been built yet for now, let's delay this logic to checkOrderByPart2.
+	} else {
+		// check only-full-group-by 4 order-by clause without group-by clause
```

Comment:
The following check isn't used for `only_full_group_by`, just for basic name resolving checks?